prov/util/src/util_attr.c

```diff
@@ -1089,6 +1089,10 @@ static void fi_alter_domain_attr(struct fi_domain_attr *attr,
 				FI_MR_BASIC : FI_MR_SCALABLE;
 	} else {
 		if ((hints_mr_mode & attr->mr_mode) != attr->mr_mode) {
+			if (attr->mr_mode & FI_MR_BASIC) {
+				attr->mr_mode &= ~FI_MR_BASIC;
+				attr->mr_mode |= FI_MR_ALLOCATED | FI_MR_PROV_KEY | FI_MR_VIRT_ADDR;
```

Comment:
The provider should indicate what mr_mode bits it supports.  If the app requests BASIC, then only BASIC may be returned, not the individual bits, which the app may not know how to interpret.

Reply:
Yeah, this is a requirement and the util code is doing the right thing by unsetting the bits that a provider doesn't require.

Reply:
@rajachan - I didn't completely follow your comment.  Are you saying that the current code is correct?

Reply:
I was agreeing with you that providers should set the bits they require and the util code should not have to do what this PR is doing. I haven't looked at the shm/uber test failure that necessitated this change yet,  but looks like that could be a provider/test bug.